### PR TITLE
fix(categoryOptionGroup): remove dataDimension field

### PR DESCRIPTION
**Pull request by Chisomchima**
this is an attempt to resolve [this](https://dhis2.atlassian.net/browse/DHIS2-18440?focusedCommentId=205831) feedback from testing of [this](https://dhis2.atlassian.net/browse/DHIS2-18440) ticket

### CODE REVIEW

File: src/components/sectionList/listView/useModelListView.tsx [hunk 1]
```diff
@@ -148,7 +148,9 @@ export const useModelListView = () => {
     const selectedView = query.data || defaultView
 
     const columns = selectedView.columns
-    const filters = selectedView.filters
+    const filters = query.data?.filters?.length
+        ? query.data.filters
+        : defaultView.filters
```
Comment (Birkbjo) [inline src/components/sectionList/listView/useModelListView.tsx:153]: Why is this necessary? In a recent PR I made it possible to remove all filters, and we shouldn't render default if a user deliberately deselects all filters?

**Comment by Birkbjo** [comment]
Looks good! We should also remove `dataDimension` from `fieldFilters` in `categoryOptionGroups/Edit.tsx`.

File: src/components/sectionList/listView/useModelListView.tsx (hunk 1 above)
Comment (Chisomchima) [reply src/components/sectionList/listView/useModelListView.tsx:153]: So `query.data.filters` shows an empty array in categoryOptionGroup while `defaultView.filters` contains the the "dymensionType" filter in the array, hence the logic added.

Comment (Birkbjo) [reply src/components/sectionList/listView/useModelListView.tsx:153]: But you would just have to select it in `ManageView`? This just means that user has done changes to the view?

**Review by Birkbjo — APPROVED** [review]

**Comment by Birkbjo** [comment]
As far as I can see `dataDimensionType` is already a filter? So I think this should just be called `fix(categoryOptionGroup): remove dataDimension field`.
